app/roles/virtual-machine/tasks/main.yml

```diff
@@ -18,10 +18,60 @@
     --cpu Haswell-noTSX
     --noautoconsole
     --check disk_size=off
-  when: inventory_hostname not in all_vms.list_vms
+    {% if virtual_machine_autostart %}--autostart{% endif %}
+  when: virtual_machine_name not in all_vms.list_vms
   delegate_to: "{{ virtual_machine_hypervisor }}"
   become: true
-
   # Haswell CPU specification fixes some bug in kvm where it might not like
   # the EL8000 CPUs. Or something. I don't know, I'm tired and it works.
   # https://bugzilla.redhat.com/show_bug.cgi?id=1657738
+
+- name: gather current ram allocation
+  shell: >
+    virsh dumpxml --inactive {{ virtual_machine_name }} |
```

Comment:
why don't we use the virt module instead of shell here?
https://docs.ansible.com/ansible/latest/collections/community/libvirt/virt_module.html

Reply:
Honestly.... just inertia. 

In a perfect world, the virt module could be used to idempotently create the vm and manage the resources. Since it doesn't (to my knowledge), I just didn't use it. Switching back and forth between the module and shell would break my brain.